jep/0000/README.adoc

```diff
@@ -0,0 +1,274 @@
+** Where feasible, usages of Guava in plugins are replaced with Java Platform equivalents.
+
+=== Rollout plan
+
+In mostly chronological order, though many tasks can be parallelized:
+
+* Prepare, merge, and release patches to remove unnecessary usages of Guava from core.
+* Prepare, merge, and release patches to remove the Guava dependency from Stapler and upgrade the Stapler dependency in core to the new version.
+* Prepare the https://github.com/jenkinsci/jenkins/pull/5707[patch to upgrade Guava in core], verifying that it passes automated tests on both Java 8 and 11.
+* Interactively verify that core and all plugins mentioned in the setup wizard (even if not `suggested`) seem to work.
+* Run the https://github.com/jenkinsci/bom[plugin BOM] test suite against the patch to upgrade Guava in core.
+* Create a https://issues.jenkins.io/browse/JENKINS-65988[Jira epic] associated with this JEP tracking all known plugins that might be affected by this change and their current status.
+* Run the https://github.com/jenkins-infra/usage-in-plugins[tool to search for usages in Jenkins plugins], checking all plugins in the update center for usages of removed APIs and classifying all matches in the https://issues.jenkins.io/browse/JENKINS-65988[Jira epic].
+* Prepare patches to make plugins compatible with both Guava 11.0.1 and the latest version of Guava, and get them merged and released.
+* Define a Jira label for regressions suspected to be related to this migration for tracking purposes.
+* Warn users of the upcoming changes on the https://www.jenkins.io/mailing-lists/[users’ mailing list], via blog post, social media, etc.
+* Merge and release the patch to upgrade Guava in core, including a warning in the https://www.jenkins.io/changelog/[release notes] and https://www.jenkins.io/doc/upgrade-guide/[upgrade guide] about the risk.
+* For a reasonable period of time (months?), monitor Jira for reported regressions as well as the overall score given to Jenkins weekly releases.
+* Track the status of “long-tail” plugins, offering advice and assistance to maintainers.
+
+== Motivation
+
+=== False positives from scanners
+
+Many security-conscious organizations using, or planning to use, Jenkins run off-the-shelf security scanners to look for known vulnerabilities.
+These commonly flag the extremely old Guava library as susceptible to a https://github.com/google/guava/wiki/CVE-2018-10237[serialization-related vulnerability] and recommend upgrading.
+While Jenkins uses https://github.com/jenkinsci/jep/tree/master/jep/200[JEP-200] to form an explicit list of allowed classes for deserialization, and the affected Guava classes are not and will never be added to the list, it is time-consuming for the https://www.jenkins.io/security/team/[security team] to respond to purported security reports and for users to justify exemptions from policy to use Jenkins anyway.
```

Comment:
To be clear, some Guava classes _are_ in our list https://github.com/jenkinsci/jenkins/blob/2cf5a80da0dd40ba1aaecd1c0fb71a38bfa131d4/core/src/main/resources/jenkins/security/whitelisted-classes.txt#L1-L28 but I guess you mean none of the ones mentioned in deserialization CVEs.

BTW https://asciidoctor.org/docs/asciidoc-recommended-practices/#one-sentence-per-line makes for easier review. (Technically this is all one sentence, but practically speaking it has several long independent clauses.)

Reply:
Sure, I'll clarify the language around "affected Guava classes" to explicitly mention the deserialization CVE.